pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/openshift/origin/pkg/client"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// NotFoundErr is returned when the imageStreamTag (ist) of interest cannot
+// be found in the graph. This doesn't mean though that the ist does not
+// exist. A user may have an image stream without a build configuration
+// pointing at it. In that case, the ist of interest simply doesn't have
+// other dependant ists
+type NotFoundErr string
+
+func (e NotFoundErr) Error() string {
+	return fmt.Sprintf("couldn't find image stream tag: %q", string(e))
+}
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	namespaces []string
+	output     string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, namespaces []string, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, namespaces: namespaces, output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace(s)
+func (d *ChainDescriber) MakeGraph(namespaces []string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{}
+	for _, namespace := range namespaces {
+		glog.V(4).Infof("Loading build configurations from %q", namespace)
+		loaders = append(loaders, &bcLoader{namespace: namespace, lister: d.Interface})
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+
+	return g, nil
+}
+
+// Describe returns the output of the graph starting from the provided
+// image stream tag (name:tag)
+func (d *ChainDescriber) Describe(namespace, name, tag string) (string, error) {
+	g, err := d.MakeGraph(d.namespaces)
+	if err != nil {
+		return "", err
+	}
+
+	// Retrieve the imageStreamTag node of interest
+	ist := g.Find(imagegraph.ImageStreamTagNodeName(imagegraph.MakeImageStreamTagObjectMeta(namespace, name, tag)))
+	if ist == nil {
+		return "", NotFoundErr(fmt.Sprintf("%s:%s", name, tag))
+	}
+
+	partitioned, err := partition(g, ist)
+	if err != nil {
+		return "", err
+	}
+
+	switch d.output {
+	case "dot":
+		data, err := dot.Marshal(partitioned, name, "", "  ", false)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	case "":
+		return humanReadableOutput(partitioned, ist), nil
+	}
+	return "", fmt.Errorf("unknown specified format %q", d.output)
+}
+
+// partition the provided graph down to a subgraph starting from the given
+// image stream tag
+func partition(g osgraph.Graph, ist graph.Node) (osgraph.Graph, error) {
```

Comment:
open an issue for a tech-debt refactor of `pkg/api/graph` create an item pointing here as a potential for promotion.